### DISCUSSION TITLE
Planning for repack releasing of "ASMBXT: Prelude To The Stupid" by Talkhaus

### DISCUSSION THREAD

**Discussion by Wohlstand** [Wohlstand/TheXTech#247] [Game projects]
### Hello!

I keep this idea for a while. I want to just put all my thinks here and make an ability to discuss anything related.

### So, what's exactly "**ASMBXT: Prelude To The Stupid**"? 
This is the major game released by the Talkhaus community around 2013 year. (there are [doze of archived discussions](https://talkhaus.raocow.com/viewforum.php?f=29) can be found here). It's the first game of the A2XT series based on the SMBX (previously there was SMW ROM hacks released under "A Super Mario Thing" title). This is the game that inspired me to give the ability to Moondust Project to make brand-new games, not relayed to SMBX. As the game works on a pure-vanilla engine, this is the only A2XT-related game that works on TheXTech natively yet (the next game called "A2MBXT, Episode 1: Analog Funk" requires the LunaDLL Autocode support and have some built-in complex level-specific codes to alter the workflow).

**So, until we can release the repack of the PTTS, we need to complete the next things at TheXTech and some other minor cases:**
- [x] Ability to rename cheat codes by gameinfo.ini (the file will be pre-included with the assets pack). This is required as some levels do require players to cheat, and not renamed cheats will confuse new players.
- [x] Ability to type cheats at non-PC platforms (using touchscreen keyboard or game controller) (done by @ds-sloth at the abstract controls branch which going to be merged with the main. Check this item after branch gets merged into the mainstream).
- [ ] Implement the standalone passwords system executor with the giving of the reward video to watch. This was a significant feature of this game that was died as the passwords system was hosted on the Talkhaus's side server that was shut down with the removal of its content. However, I do have all stuff given by **Kashkabald** who originally maintained the passwords system and explained how it actually worked. I'll try to make the sort of monolithic HTML page that could serve as a passwords system standalone (all reward content will be encrypted and can only be decrypted by one of the typed in-game passwords).
- [ ] Replace all existing music with a new one by [this list](https://asmbxt.fandom.com/wiki/Prelude_to_the_Stupid_music_replacement) (we built it together with SAJewers to get rid of all music that caused strikes of copyrusters at YouTube videos. However, the update was applied to another copy of PTTS we had, which is, looks, slightly broken, and I need to redo all work from the scratch again).

- [x] **Optionally and not required**, implement the built-in deaths counter (keep it disabled by default) as a replacement for the standalone hacking tool that was shipped with the original game to trace player deaths. Later, the deaths counter became an embedded feature of the LunaDLL that was used with the "A2MBXT, Episode 1: Analog Funk"

- [ ] **More long-term task**: cooperate with Talkhaus community to clear up the game from a doze of unwanted stuff. Because this term is REQUIRED, I can't release this game by myself independently. So, instead of that, I could release the manual and the patch to let users on their ends patch the original copy of the game to work with TheXTech natively.

**Comment by LoveBodhi** [comment]
https://freepd.com/

We might use FreePD Musics...

**Comment by Wohlstand** [comment]
Deaths counter has been implemented at `wip-memory-emulator` branch
